File: Recomp_Resource/client/src/components/message/Message.js
const Message = ({ message }) => {
  return (
    <tr>
      <th scope="row">{message.opened === true ? "Opened" : "New"}</th>
      <td>
        <a href={`../../user/details/${message.senderId}`}>
          <strong>{message?.sender?.displayName}</strong>
        </a>
      </td>
      <td>
        <a href={`../../message/details/${message.id}`}>
          <span>{message.subject}</span>
        </a>
      </td>
      <td>{message.dayCreated}</td>
    </tr>
  );
};
export default Message;

File: Recomp_Resource/Models/Message.cs
using System;

namespace Recomp_Resource.Models
{
    public class Message
    {
        public int Id { get; set; }
        public string Subject { get; set; }
        public DateTime DateCreated { get; set; }
        public string DayCreated
        {
            get
            {
                string Days = "";
                TimeSpan daysPast = DateTime.Now - DateCreated;
                if (daysPast.Days < 1)
                {
                    Days = DateCreated.ToString("hh:mm tt");
                }
                else if(daysPast.Days < 7 )
                {
                    Days= DateCreated.DayOfWeek.ToString();
                } else
                {
                    Days = DateCreated.Date.ToString("MM/dd/yyyy");
                }
                return Days;
            }
        }
        public bool Opened { get; set; }

        public int SenderId { get; set; }
        public User Sender { get; set; }

        public int RecipientId { get; set; }
        public User Recipient { get; set; }
        public string Content { get; set; }

        public int NewMessages { get; set; }

        public bool RecipientHidden { get; set; }

        public bool SenderHidden { get; set; }
    }
}
